bin/check-marathon-apps.rb

```diff
@@ -0,0 +1,398 @@
+    end
+
+    # Get Marathon API apps
+    apps = fetch_apps
+
+    # Get Marathon API queue
+    queue = fetch_queue
+
+    # Get and parse default check config
+    check_config_str = if !config[:default_check_config].nil?
+                         config[:default_check_config]
+                       elsif !config[:default_check_config_file].nil?
+                         File.read(config[:default_check_config_file])
+                       else
+                         DEFAULT_CHECK_CONFIG
+                       end
+    check_config = parse_json(check_config_str)
+
+    # Filter apps, if both exists exclude pattern will override match pattern
+    apps.keep_if { |app| app['id'][/#{config[:match_pattern]}/] } if config[:match_pattern]
+    apps.delete_if { |app| app['id'][/#{config[:exclude_pat]}/] } if config[:exclude_pat]
+
+    apps.each do |app|
+      # Select app queue if any
+      app_queue = queue.select { |q| q['app']['id'][/^#{app['id']}$/] }.to_a.first
+
+      # Build check result
+      check_result = check_result_scaffold(app)
+
+      # Parse Marathon app labels
+      labels_config = parse_app_labels(app['labels'].to_h)
+
+      REFERENCES.each do |reference|
+        # / is and invalid character
+        check_result['name'] = "check_marathon_app#{app['id'].tr('/', '_')}_#{reference}"
+
+        state = case reference
+                when 'health'
+                  get_marathon_app_health(app)
+                when 'status'
+                  get_marathon_app_status(app, app_queue.to_h)
+                end
+
+        # Merge user provided check config
+        check_result.merge!(check_config.dig('_').to_h)
+        check_result.merge!(check_config.dig(reference, '_').to_h)
+        check_result.merge!(check_config.dig(reference, state).to_h)
+
+        # Merge Marathon parsed check config
+        check_result.merge!(labels_config.dig('_').to_h)
+        check_result.merge!(labels_config.dig(reference, '_').to_h)
+        check_result.merge!(labels_config.dig(reference, state).to_h)
+
+        # Build check result output
+        check_result['output'] = "#{reference.upcase} #{state.capitalize} - "\
+          "tasksRunning(#{app['tasksRunning'].to_i}), tasksStaged(#{app['tasksStaged'].to_i}), "\
+          "tasksHealthy(#{app['tasksHealthy'].to_i}), tasksUnhealthy(#{app['tasksUnhealthy'].to_i})"
+
+        # Make sure that check result data types are correct
+        sanitize_check_result(check_result)
+
+        # Send the result to sensu-client HTTP socket
+        post_check_result(check_result)
+      end
+    end
+
+    ok 'Marathon Apps Status and Health check is running properly'
+  end
+
+  def check_result_scaffold(app)
+    {
+      'name' => '',
+      'executed' => Time.now.to_i,
+      'marathon' => app.select { |k, _| config[:marathon_keys].split(',').include?(k) },
+      'source' => 'marathon',
+      'output' => '',
+      'status' => 3
+    }
+  end
+
+  def sanitize_check_result(check_result)
```

Comment:
this function feels like it could be more appropriately named `enforce_sensu_field_types` or something as when I first saw it I assumed it was more along the lines of putting in some redaction.

Reply:
thanks for the suggestion, updating as you suggest.